### PR TITLE
Leave nonexistent mappings empty instead of inserting garbage data

### PR DESCRIPTION
**Pull request by natanfudge**

### CODE REVIEW

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ private void writeClass(PrintWriter writer, EntryTreeNode<EntryMapping> node, En
 			if (mapping != null) {
 				parts.addFirst(mapping.getTargetName());
 			} else {
-				parts.addFirst(classEntry.getFullName());
```
Comment (liach) [inline src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java:62]: Why not just omit the call

**Review by liach — CHANGES_REQUESTED** [review]

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java (hunk 1 above)
Comment (natanfudge) [reply src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java:62]: wouldn't that mean we lose the trailing tab?

Comment (liach) [reply src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV2Writer.java:62]: Even when you have tabs, this means you are literally declaring `""` as the name in those trailing namespaces, which would make the thing worse

**Comment by liach** [comment]
Turns out that you are using the tiny v2 mappings incorrectly. Those earlier entries are not "trash".

**Comment by natanfudge** [comment]
They are not "trash", they are intermediary names that are being put in the named column because there is no named mapping. In this case the value should just be empty to signify there is no mapping, not insert the intermediary name. This insertion of intermediary name when there is no named mapping is what I called "garbage data".

**Comment by liach** [comment]
d05485b7aa8c5571c9a130db112dc9ee89483c9b and 8ed5cb9d6c2a1e7c9376639a60fc6234bdd258e2 should fix your issues.

**Pull request closed without merging**